test/common.sh

```diff
@@ -0,0 +1,260 @@
+  # failure in kubetest.
+  # TODO(adrcunha): Get rid of this workaround.
+  echo -n "$1"> ${TEST_RESULT_FILE}
+}
+
+function detect_changed_e2e_test() {
+    # check if any file from test/ directory is changed
+    echo ${CHANGED_FILES} | grep "test/" || true
+}
+
+function print_changed_resources_to_console() {
+    # detect for changes in tests dir of the task
+    echo ${CHANGED_FILES} | grep -o 'task/[^\/]*/[^\/]*/tests/[^/]*'|xargs -I {} dirname {}|sed 's/\(tests\).*/\1/g'
+    # detect for changes in the task manifest
+    echo ${CHANGED_FILES} | grep -o 'task/[^\/]*/[^\/]*/*[^/]*.yaml'|xargs -I {} dirname {}|awk '{print $1"/tests"}'
+}
+
+# Signal (as return code and in the logs) that all E2E tests passed.
+function success() {
+  set_test_return_code 0
+  echo "**************************************"
+  echo "***        E2E TESTS PASSED        ***"
+  echo "**************************************"
+  exit 0
+}
+
+function show_failure() {
+    local testname=$1 test_namespace=$2
+
+    echo "FAILED: ${testname} task has failed to comeback properly" ;
+    ${KUBECTL_CMD} api-resources
+    echo "Namespace: ${test_namespace}"
+    echo "--- TaskRun Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${test_namespace} taskrun -o yaml
+    echo "--- Task Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${test_namespace} task -o yaml
+    echo "--- PipelineRun Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${test_namespace} pipelinerun -o yaml
+    echo "--- Pipeline Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${test_namespace} pipeline -o yaml
+    echo "--- StepAction Dump"
+    ${KUBECTL_CMD} get --ignore-not-found=true -n ${test_namespace} stepaction -o yaml
+    echo "--- Container Logs"
+    for pod in $(${KUBECTL_CMD} get pod -o name -n ${test_namespace}); do
+        echo "----POD_NAME: ${pod}---"
+        ${KUBECTL_CMD} logs --all-containers -n ${test_namespace} ${pod} || true
+    done
+    exit 1
+}
+
+function test_yaml_can_install() {
+    # Validate that all the Task CRDs in this repo are valid by creating them in a NS.
+    ns="test-yaml-ns"
+    all_tasks="$*"
+    ${KUBECTL_CMD} create namespace "${ns}" || true
+
+    local runtest
+    for runtest in ${all_tasks}; do
+        # remove task/ from beginning
+        local runtestdir=${runtest#*/}
+        # remove /0.1/tests from end
+        local testname=${runtestdir%%/*}
+        runtest=${runtest//tests}
+
+        # in case a task is being removed then it's directory
+        # doesn't exists, so skip the test for YAML
+        [ ! -d "${runtest%%/*}/${testname}" ] && continue
+
+        runtest="${runtest}${testname}.yaml"
+        skipit=
+        for ignore in ${TEST_YAML_IGNORES}; do
+            [[ ${ignore} == "${testname}" ]] && skipit=True
+        done
+
+        # don't test the tasks which are deprecated
+        cat ${runtest} | grep 'tekton.dev/deprecated: \"true\"' && skipit=True
```

Comment:
We use the `build.appstudio.redhat.com/expires-on` label (but its value can be in the future)